aggsender/db/aggsender_db_storage.go

```diff
@@ -86,17 +106,18 @@ var _ AggSenderStorage = (*AggSenderSQLStorage)(nil)
 
 // AggSenderSQLStorageConfig is the configuration for the AggSenderSQLStorage
 type AggSenderSQLStorageConfig struct {
-	DBPath                  string
-	CertificatesDir         string
-	KeepCertificatesHistory bool
+	DBPath                   string
+	CertificatesDir          string
+	RetainCertificatesPolicy StorageRetainCertificatesPolicy
```

Comment:
WDYT to use interface here as well? It seems to me that it is doable, because this struct is not used to unmarshal the TOML values to it. WDYT?
```suggestion
	RetainCertificatesPolicy StorageRetainCertificatesPolicier
```

Reply:
I'm on the fence. This struct is a config, so I expect to have key value, not the behaviour (interface) that is not used.  By other hand in the object  `AggSenderSQLStorage` I want the behaviour, so I use the interface. WDYT?

Reply:
I'm fine with either, the reason why I've suggested it, is because we:
- return a pointer when constructing the default retain policy [here](https://github.com/agglayer/aggkit/blob/5fba9e20677a8ebe051b15e32dc1bec28b9966cc/aggsender/db/retain_certificates_policy.go#L65-L70)
- then we do the [de-referencing](https://github.com/agglayer/aggkit/blob/5fba9e20677a8ebe051b15e32dc1bec28b9966cc/aggsender/validator/validate_db.go#L37)

I don't see a reason to do it. If it needs to be a value type, then constructor function should not return a pointer.

If we switch to use the interface, then we are consistent (as we don't need to de-reference the pointer). WDYT?